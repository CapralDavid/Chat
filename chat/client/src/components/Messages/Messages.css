.messages {
  padding: 5% 0;
  overflow: auto;
  flex: auto;
}
.messages:after {
  content: '';
  position: absolute;
  display: block;
  width: 0;
  z-index: 1;
  border-style: solid;
  border-width: 0 0 39px 32px;
  border-color: transparent transparent #000000 transparent;
  top: 66%;
  left: -32px;
  margin-top: -19.5px;
 }