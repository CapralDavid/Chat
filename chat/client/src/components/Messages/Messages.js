import React from 'react';

import './Messages.css';
import ScrollToBottom from  'react-scroll-to-bottom';

import Message from './Message/Message.js'

//СДЕЛАТЬ CSS




let Messages = ({ messages, name }) => (
    <ScrollToBottom className = "messages">
         {messages.map((message, i) => <div key={i}> <Message message={message} name={name}/> </div>)}

    </ScrollToBottom>
);

//{messages.map((message, i) => <div key={i}> <Message message={message} name={name}/> </div>)}
//заменил <Message /> на просто key и все избражается
//проблема в том что ?массив messages пустой?
//!!!!проверь файл message.js


// let Messages = ({messages, name }) => (
//     <ScrollToBottom>
//          {messages.map((message, i) => <div key={i}> <Message message={message} name={name}/> </div>)}

//     </ScrollToBottom>
// );




export default Messages;