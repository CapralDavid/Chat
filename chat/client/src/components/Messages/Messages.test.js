import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Messages from './Messages.js';


//без обновления снапшота(npm test -- -u) элементы с react-scroll-to-bottom
//не совпадают,так как сощдавая новый элемент класс изменяется,хотя 
//деле ничего не изменилось
//просто каждый раз сайт создает новый элемент

describe('test render Join element', () => {

    test('render correct: Messages Empty', () => {
        const varOne = []
        const nameOne = ""
          const component = renderer.create(
              <Router>
              <Messages messages={varOne} name={nameOne}></Messages>
              </Router>
            );
        
          let treeMessages = component.toJSON();
          expect(treeMessages).toMatchSnapshot();
        
        }); 



    test('render correct: Messages One', () => {
    const varOne = [{user:"one", text:"hello its one"}]
    const nameOne = "one"
      const component = renderer.create(
          <Router>
          <Messages messages={varOne} name={nameOne}></Messages>
          </Router>
        );
    
      let treeMessages = component.toJSON();
      expect(treeMessages).toMatchSnapshot();
    
    }); 


    test('render correct: Messages several', () => {
        const varOne = [{user:"one", text:"hello its one"}, {user:"two", text:"hello its two"}]
        const nameOne =  "one two"
          const component = renderer.create(
              <Router>
              <Messages messages={varOne} name={nameOne}></Messages>
              </Router>
            );
        
          let treeMessages = component.toJSON();
          expect(treeMessages).toMatchSnapshot();
        
        }); 
      
  
  
  
  
  });