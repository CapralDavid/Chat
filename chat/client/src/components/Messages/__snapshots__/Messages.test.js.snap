// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render Join element render correct: Messages Empty 1`] = `
<div
  className="react-scroll-to-bottom--css-ahwli-79elbk messages"
>
  <div
    className="react-scroll-to-bottom--css-ahwli-1n7m0yu"
  />
</div>
`;

exports[`test render Join element render correct: Messages One 1`] = `
<div
  className="react-scroll-to-bottom--css-ahwli-79elbk messages"
>
  <div
    className="react-scroll-to-bottom--css-ahwli-1n7m0yu"
  >
    <div>
       
      <div
        className="msgContainer justifyEnd"
      >
        <p
          className="sentMsg pr-10"
        >
          one
        </p>
        <div
          className="msgBox"
        >
          <p
            className="msgText colorWhite"
          >
            hello its one
          </p>
        </div>
      </div>
       
    </div>
  </div>
</div>
`;

exports[`test render Join element render correct: Messages several 1`] = `
<div
  className="react-scroll-to-bottom--css-ahwli-79elbk messages"
>
  <div
    className="react-scroll-to-bottom--css-ahwli-1n7m0yu"
  >
    <div>
       
      <div
        className="msgContainer justifyStart"
      >
        <div
          className="msgBox"
        >
          <p
            className="msgText colorDark"
          >
            hello its one
          </p>
        </div>
        <p
          className="sentMsg pl-10"
        >
          one
        </p>
      </div>
       
    </div>
    <div>
       
      <div
        className="msgContainer justifyStart"
      >
        <div
          className="msgBox"
        >
          <p
            className="msgText colorDark"
          >
            hello its two
          </p>
        </div>
        <p
          className="sentMsg pl-10"
        >
          two
        </p>
      </div>
       
    </div>
  </div>
</div>
`;
